src/conn/conn_sweep.c

```diff
@@ -13,6 +13,41 @@
       __wt_atomic_loadi32(&(dhandle)->session_inuse) == 0 &&      \
       __wt_atomic_load32(&(dhandle)->references) == 0)
 
+/*
+ * __sweep_check_file_handle_exists --
+ *     Check if the file dhandle exists for the table dhandle.
+ */
+static int
+__sweep_check_file_handle_exists(WT_SESSION_IMPL *session, WT_DATA_HANDLE *dhandle)
+{
+    WT_DECL_RET;
+    WT_TABLE *table;
+
+    /*
+     * The sweep server's algorithm is altered to prevent unnecessary table dhandle closures. This
+     * is done by checking for associated file dhandles before marking table dhandles for sweeping.
+     * It resolves schema lock contention caused by repetitive table dhandle operations during
+     * MongoDB cursor activity on simple tables, and ensures table dhandles are retained for active
+     * file dhandles, which is required for file dhandle access.
+     */
+    table = (WT_TABLE *)dhandle;
+    if (table->is_simple && table->cg_complete) {
+        ret = __wt_conn_dhandle_find(session, table->cgroups[0]->source, NULL);
+
+        /*
+         * Reset the time of death if the file dhandle exists for the associated table dhandle.
+         */
+        if (ret == 0) {
+            dhandle->timeofdeath = 0;
+            return (1);
```

Comment:
My suggestion here is to return (0) as a success. And if the file is not found, return WT_NOTFOUND. Assign the ret with WT_NOTFOUND by default.

In the caller, if it is success `ret == 0`, continue instead `ret == 1`. Other than WT_NOTFOUND is an error with WT_RET_NOTFOUND_OK(ret) check.

Reply:
Fixed in 5a6b14ac5bcbefd9d77645ff1337a30c768541e7